mlflow/protos/service.proto

```diff
@@ -707,6 +723,110 @@ message ListExperiments {
   }
 }
 
+message SearchExperiments {
+  option (scalapb.message).extends = "com.databricks.rpc.RPC[$this.Response]";
+
+  // Qualifier for type of experiments to be returned.
+  // If unspecified, return only active experiments.
+  optional ViewType view_type = 1;
+
+  // Maximum number of experiments desired.
+  // Servers may select a desired default `max_results` value. All servers are
+  // guaranteed to support a `max_results` threshold of at least 1,000 but may
+  // support more. Callers of this endpoint are encouraged to pass max_results
+  // explicitly and leverage page_token to iterate through experiments.
+  optional int64 max_results = 2;
+
+  // Pagination token to go to the next page based on a previous query.
+  optional string page_token = 3;
+
+  optional string filter = 4;
+
+  repeated string order_by = 5;
+
+  message Response {
+    // All experiments.
+    repeated Experiment experiments = 1;
+
+    // Pagination token to request next page of experiments for the same query.
+    optional string next_page_token = 2;
+  }
+}
+
+message SearchExperimentsDatabricks {
+  option (scalapb.message).extends = "com.databricks.rpc.RPC[$this.Response]";
+  option (scalapb.message).extends = "com.databricks.mlflow.api.MlflowTrackingMessage";
+
+  // Maximum number of experiments desired. Max threshold is 3000.
+  // NOTE: the frontend will initially always override this parameter to 3000,
+  // but we include it so that we can adjust the number of max results fetched
+  // via static feature flag in the MLflow service as needed for stability purposes
+  optional int64 max_results = 1  [default = 100];
+
+  // Token indicating the page of experiments to fetch
+  optional string page_token = 2;
+
+  // String representing a SQL filter condition (e.g. "name ILIKE 'my-experiment%'")
+  optional string filter = 3;
+
+  // List of columns for ordering search results, which can include experiment name and last updated
+  // timestamp with an optional "DESC" or "ASC" annotation, where "ASC" is the default.
+  // Tiebreaks are done by experiment id DESC.
+  repeated string order_by = 4;
+
+  message Response {
+    // Experiments that match the search criteria
+    // DEPRECATED: Please use experiments_databricks instead.
+    repeated Experiment experiments = 1 [deprecated = true];
+
+    // Internal Experiments that match the search criteria.
+    // These experiments have additional edge fields,
+    // such as actions allowed to perform on the experiment.
+    repeated ExperimentDatabricks experiments_databricks = 2;
+
+    // Token that can be used to retrieve the next page of experiments.
+    // An empty token means that no more experiments are available for retrieval.
+    optional string next_page_token = 3;
+  }
+}
```

Comment:
@harupy Can we remove this Databricks-specific proto?

Reply:
Removed :)